tensorflow_addons/optimizers/cocob.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""COntinuos COin Betting (COCOB) Backprop optimizer"""
+
+from typeguard import typechecked
+from tensorflow.python.ops import state_ops
+from tensorflow.python.ops import control_flow_ops
```

Comment:
Generally we don't use private API import.  Are these required?

Reply:
Yes. It is required.

Reply:
Why? Isn't `tf.group` enough?

Reply:
Do you consider ```state_ops``` as a private api? As I used ```state_ops.assign```. 
And ```tf.group``` works fine (checked). Old habit of using ```control_flow_ops```  :-)

Reply:
Private API are e.g. `tensorflow.python.*` this will fail our CI check. We could bypass but only when strictly needed.
So if you have the public API  don't import `tensorflow.python.*`.

Reply:
I have removed the dependency of the private API's.